setup.py

```diff
@@ -12,4 +16,22 @@ def has_ext_modules(self):
         return True
 
 
-setup(version="0.45.3.dev0", packages=find_packages(), distclass=BinaryDistribution)
+class ExtBuildPy(build_py):
+    def run(self):
+        # build_cmake needs to be called prior to build_py, as the latter
+        # collects the files output into the package directory.
+        try:
+            self.run_command("build_cmake")
+        except DistutilsModuleError:
+            warn("scikit-build-core not installed, CMake will not be invoked automatically. "
+                 "Please install scikit-build-core or run CMake manually to build extensions.")
+        super().run()
+
+
+setup(version="0.45.3.dev0",
+      packages=find_packages(),
+      distclass=BinaryDistribution,
```

Comment:
Technically, this is no longer necessary when CMake is called. However, since I preserved the support for using `setup.py` without `scikit-build-core` installed, I've left this hack in.